### PR TITLE
Added getID to PublicKey

### PR DESCRIPTION
**Pull request by braydonf**
- Tests in BloomFilter in [bitcoin core](https://github.com/bitcoin/bitcoin/blob/master/src/test/bloom_tests.cpp#L85) use the CPubKey.GetID as input into the BloomFilter
- The hash is calculated when generating an Address however the hash itself is not currently available

### CODE REVIEW

File: lib/publickey.js [hunk 1]
```diff
@@ -384,6 +385,15 @@ PublicKey.prototype.toBuffer = PublicKey.prototype.toDER = function() {
...
 };
 
+/*
+ * Will return a sha256 + ripemd160 hash of the serialized public key
+ * @see https://github.com/bitcoin/bitcoin/blob/master/src/pubkey.h#L141
+ * @returns {Buffer}
+ */
+PublicKey.prototype.getID = function getID() {
+  return Hash.sha256ripemd160(this.toBuffer())
+}
```
Comment (eordano) [inline lib/publickey.js]: missing `;`

File: lib/publickey.js [hunk 2]
```diff
@@ -384,6 +385,15 @@ PublicKey.prototype.toBuffer = PublicKey.prototype.toDER = function() {
   }
 };
 
+/*
+ * Will return a sha256 + ripemd160 hash of the serialized public key
+ * @see https://github.com/bitcoin/bitcoin/blob/master/src/pubkey.h#L141
+ * @returns {Buffer}
+ */
+PublicKey.prototype.getID = function getID() {
```
Comment (eordano) [inline lib/publickey.js]: Can we rename this to `fingerprint`, `hash`, or `hash160` and/or make this start with `_`? I feel strange about "the ID of a public key"

Comment (braydonf) [inline lib/publickey.js]: Hash has an entirely different meaning of a double sha256 hash, see: https://github.com/bitcoin/bitcoin/blob/master/src/pubkey.h#L148

File: lib/publickey.js [hunk 3]
```diff
@@ -384,6 +385,15 @@ PublicKey.prototype.toBuffer = PublicKey.prototype.toDER = function() {
   }
 };
 
+/*
```
Comment (eordano) [inline lib/publickey.js]: Missing extra `*`

File: lib/publickey.js (hunk 2 above)
Comment (eordano) [inline lib/publickey.js]: I see. Still hate the name, but it's not that bad as it's the name in the ref implementation.

**Comment by eordano** [comment]
LGTM after missing `*` on jsdoc

**Pull request merged**